### ISSUE TITLE
Enable Top Level Await

### ISSUE THREAD

**Issue by mudgen** [eslint/eslint#13178]
**Please describe what the rule should do:**
Provide a setting or rule that enables top level await in code without an error or warning.

V8 now supports top level await:  https://v8.dev/features/top-level-await

As a standard top level await is in stage 3: https://github.com/tc39/proposal-top-level-await

I am currently using deno which has top level await,  so this new setting/rule would be very useful for me.

**Comment by mdjermanovic** [comment]
Hi @mudgen, thanks for the issue!

I assume you're getting parsing errors. ESLint's default parser will support this syntax if and when it reaches stage 4, per our [policy about experimental features](https://github.com/eslint/eslint#what-about-experimental-features).

In the meantime, you can use `babel-eslint` parser.

**Comment by mjsalinger** [comment]
@mdjermanovic - We don't use babel, but node.js just added support for top-level await.  What is the recommended course if we want to use this feature of node without using Babel?

**Comment by anikethsaha** [comment]
top level await is still in [stage 3](https://github.com/tc39/proposal-top-level-await).

eslint supports stage 4 proposal.

**Comment by mdjermanovic** [comment]
@mjsalinger our default parser supports only stage 4 proposals. You could use `babel-eslint` as a parser for eslint. without using Babel to transpile the code.

**Comment by billinghamj** [comment]
Although it is still in stage 3, it has now shipped unflagged in Node and other common JS environments, so might be worth reconsidering

**Comment by boggddan** [comment]
> @mdjermanovic - We don't use babel, but node.js just added support for top-level await. What is the recommended course if we want to use this feature of node without using Babel?

Hi, Could help me?
I install  "@babel/eslint-parser" and  "@babel/plugin-syntax-top-level-await", but VS Code write error: **Parsing error: Cannot use keyword 'await' outside an async function**

![2020-09-21_20-46](https://user-images.githubusercontent.com/24915025/93803413-7d1dae80-fc4d-11ea-995b-75cdc0f1aead.png)

package.json
```json
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "fastify": "^3.4.1"
  },
  "devDependencies": {
    "@babel/core": "^7.11.6",
    "@babel/eslint-parser": "^7.11.5",
    "@babel/plugin-syntax-top-level-await": "^7.10.4",
    "eslint": "^7.9.0",
    "eslint-config-airbnb-base": "^14.2.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-node": "^11.1.0"
  }
}
```

.eslintrc.cjs
```javascript
module.exports = {
  parserOptions: {
    ecmaVersion: 2021,
    sourceType: 'module',
    parser: "@babel/eslint-parser"
  }
};
```

babel.config.json
```json
{
  "plugins": ["@babel/plugin-syntax-top-level-await"]
}
```

Thank's

**Comment by mdjermanovic** [comment]
Hi @boggddan!

`parser` should be at the top level, please try with this `.eslintrc.cjs`:

```js
module.exports = {
  parser: "@babel/eslint-parser",
  parserOptions: {
    ecmaVersion: 2021,
    sourceType: 'module'
  }
};
```

**Comment by iwaduarte** [comment]
+1 for enabling top level await even if it is not stage 4. Not everything is black and white. We should not be too constrained to rules.  

Also another option solution without using babel-parser would also be welcoming, like disabling this specific parse error ?

**Comment by pubmikeb** [comment]
The Top Level Await is now supported by Node.js without any flags even at 3rd stage, so it would be great to adopt this rule without waiting another 7 months until this feature will be accepted by TC39.

**Comment by ljharb** [comment]
(what is RTM?)

**Comment by pubmikeb** [comment]
> (what is RTM?)

https://en.wikipedia.org/wiki/Software_release_life_cycle#RTM
